archive/compression/compression.go

```diff
@@ -296,22 +302,22 @@ func cmdStream(cmd *exec.Cmd, in io.Reader) (io.ReadCloser, error) {
 	return reader, nil
 }
 
-func detectPigz() string {
-	path, err := exec.LookPath("unpigz")
+func detectCommand(path, disableEnvName string) string {
+	path, err := exec.LookPath(path)
 	if err != nil {
-		log.L.WithError(err).Debug("unpigz not found, falling back to go gzip")
+		log.L.WithError(err).Debugf("%s not found", path)
 		return ""
 	}
 
-	// Check if pigz disabled via CONTAINERD_DISABLE_PIGZ env variable
-	value := os.Getenv(disablePigzEnv)
+	// Check if this command is disabled via the env variable
+	value := os.Getenv(disableEnvName)
```

Comment:
Can the order of this be reversed ? , so we first check if the `disableEnvName` is set/unset.  If the command is disabled, then we dont need to do a lookup. This also prevents from logging an unwanted `not found` error , when the command is disabled.

Reply:
Makes sense. Addressed in 7f399ce.